sound/soc/sof/intel/hda-dai-ops.c

```diff
@@ -189,6 +189,17 @@ static unsigned int hda_calc_stream_format(struct snd_sof_dev *sdev,
 	return format_val;
 }
 
+static struct hdac_ext_link *hda_get_hlink(struct snd_sof_dev *sdev,
+					   struct snd_soc_dai *cpu_dai,
```

Comment:
Why do we need `cpu_dai` if it is not used?

Reply:
yes, that's true. I didn't notice it, so far all that's needed in sdev and the substream. This can be simplified, thanks for the suggestion @bardliao

Reply:
removed cpu_dai